samples/99_zil_interpreter/app/builtins.rb

```diff
@@ -7,15 +7,30 @@ def define_for_evaled_arguments(&implementation)
   }
 end
 
+# helper method to check for argument count
+def argc!(arguments, count, name, cmp_type = :==)
+  raise FunctionError, "#{name.inspect} has an arity (#{cmp_type}#{count}), got #{arguments.length}" unless arguments.length.send(cmp_type, count)
+end
```

Comment:
Good idea!
I think maybe having several more explicitly names helpers would be nice:
- `expect_argument_count!(arguments, count)`
- `expect_argument_count_in_range!(arguments, count)`
- `expect_minimum_argument_count!(arguments, count)`

That should cover all relevant cases.

Also I think we don't need the function name in the exception texts.... 
It would be cleaner to catch `FunctionErrors` inside `eval_zil` and then re-raise them with the Function name added in front....

Reply:
Also it would be good to use these new helpers in all builtins you added